### ISSUE TITLE
Identify() method identifies .doc files as .xls files

### ISSUE THREAD

**Issue by psuplat** [PHPOffice/PhpSpreadsheet#3507]
This is a bug report

### What is the expected behavior?
When using` IOFactory::identify()` correct file type should be detected

### What is the current behavior?
When user uploads office word file (.doc extension) the `\PhpOffice\PhpSpreadsheet\IOFactory::identify()` returns 'Xls' as file type. 
There is no such issue with .docx files

### What are the steps to reproduce?
upload older version of word file, wit .doc extension

```php
<?php

use \PhpOffice\PhpSpreadsheet\IOFactory;

class SomeClass {
    public function testFile() {
        $file_path = 'path/to/file/test.doc'; //fetch from file system or from upload form
        $inputFileType = IOFactory::identify($file);
        echo $inputFileType; //should return Doc, returns Xls
    }
}
```

### Which versions of PhpSpreadsheet and PHP are affected?
PhpSpreadsheet = 1.25.2
PHP = 8.1.8

**Comment by MarkBaker** [comment]
The logic only tests if it's an OLE2 file, which includes `doc`, `ppt`, `xls`, so any of those formats will match. When we wrote that code, we made the misguided assumption that (as this was a spreadsheet library) developers would only use it for spreadsheet files and not for other office document types.
We can tighten the logic so that it does additional checks to see which of those formats it is, with a degree of performance overhead; but we're never going to return `Doc`: the return value is the name of the Reader class that can be used to load the file, and as this library is for spreadsheet files, we 'don't have Readers for any other types of files.
